### PR TITLE
TestCompileAll improvements

### PR DESCRIPTION
**Pull request by nicolaspayette**
@frankduncan, quick review?

### CODE REVIEW

File: src/main/org/nlogo/headless/ChecksumsAndPreviews.scala [hunk 1]
```diff
@@ -41,6 +41,8 @@ object ChecksumsAndPreviews {
   }
 
   object Previews {
+    def needsManualPreview(previewCommands: String) =
+      previewCommands contains "need-to-manually-make-preview-for-this-model"
```
Comment (SethTisue) [inline src/main/org/nlogo/headless/ChecksumsAndPreviews.scala]: surely this is always false? should be `containsSlice`, not `contains`

two cheers for Scala's non-typesafe `contains` method

Comment (nicolaspayette) [inline src/main/org/nlogo/headless/ChecksumsAndPreviews.scala]:
```
scala> def needsManualPreview(previewCommands: String) =
     | previewCommands contains "need-to-manually-make-preview-for-this-model"
needsManualPreview: (previewCommands: String)Boolean

scala> needsManualPreview("need-to-manually-make-preview-for-this-model")
res1: Boolean = true
```
I think it is `java.lang.String.contains` being called here, not `scala.collection.SeqLike.contains`. `contains` reads clearer to me than `containsSlice`, and is probably a bit faster. Would there be any reason to be using `containsSlice` anyway? (A bit clearer, perhaps?)

Comment (SethTisue) [inline src/main/org/nlogo/headless/ChecksumsAndPreviews.scala]: hah. you are right, of course.

yes, I would argue that `containsSlice` is clearer, to Scala eyes — it doesn't lead the careful reader who knows Scala well down the garden path I went down. but then I _would_ say that, to make myself look less foolish for going down that garden path :-)

Comment (nicolaspayette) [inline src/main/org/nlogo/headless/ChecksumsAndPreviews.scala]: This code will hopefully be short-lived anyway: I have something else in store for `6.x` that should look much better to Scala eyes... :-)

**Comment by nicolaspayette** [comment]
This, by the way, should close #762.

**Pull request merged**